packages/std/src/ibc.rs

```diff
@@ -0,0 +1,163 @@
+pub struct PortIdResponse {
+    pub port_id: String,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct ListChannelsResponse {
+    pub channels: Vec<IbcEndpoint>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct ChannelResponse {
+    pub channel: IbcChannel,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct IbcEndpoint {
+    pub port_id: String,
+    pub channel_id: String,
+}
+
+// These are various messages used in the callbacks
+
+/// IbcChannel defines all information on a channel.
+/// This is generally used in the hand-shake process, but can be queried directly.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct IbcChannel {
+    pub endpoint: IbcEndpoint,
+    pub counterparty_endpoint: IbcEndpoint,
+    pub order: IbcOrder,
+    pub version: String,
+    /// CounterpartyVersion can be None when not known this context, yet
+    pub counterparty_version: Option<String>,
+    /// The connection upon which this channel was created. If this is a multi-hop
+    /// channel, we only expose the first hop.
+    pub connection_id: String,
+}
+
+// TODO: check what representation we want here for encoding - string or number
+/// IbcOrder defines if a channel is ORDERED or UNORDERED
+#[repr(u8)]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub enum IbcOrder {
+    Unordered = 1,
+    Ordered = 2,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct IbcPacket {
+    /// The raw data send from the other side in the packet
+    pub data: Binary,
+    /// identifies the channel and port on the sending chain.
+    pub src: IbcEndpoint,
+    /// identifies the channel and port on the receiving chain.
+    pub dest: IbcEndpoint,
+    /// The sequence number of the packet on the given channel
+    pub sequence: u64,
+    /// block height after which the packet times out
+    pub timeout_height: u64,
+    /// block timestamp (in nanoseconds) after which the packet times out
+    pub timeout_timestamp: u64,
+    // the version that the client is currently on
+    pub version: u64,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct IbcAcknowledgement {
+    pub acknowledgement: Binary,
+    pub original_packet: IbcPacket,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct IbcConnectResponse<T = Empty>
+where
+    T: Clone + fmt::Debug + PartialEq + JsonSchema,
+{
+    pub messages: Vec<CosmosMsg<T>>,
+    /// The attributes that will be emitted as part of a "wasm" event
+    pub attributes: Vec<Attribute>,
+}
+
+impl<T> Default for IbcConnectResponse<T>
+where
+    T: Clone + fmt::Debug + PartialEq + JsonSchema,
+{
+    fn default() -> Self {
+        IbcConnectResponse {
+            messages: vec![],
+            attributes: vec![],
+        }
+    }
+}
```

Comment:
Yeah, same as the rest of them: https://github.com/CosmWasm/cosmwasm/blob/e2ea1990c411420ccf885a788275d22d2235684a/packages/std/src/results/handle.rs#L23-L34

Maybe this works now with a newer rust compiler?

Reply:
Still produces the following error (when I try to auto-derive it for handle repsonse):

```
error[E0277]: the trait bound `types::Empty: std::default::Default` is not satisfied
  --> packages/std/src/results/contract_result.rs:93:73
   |
93 |         let result: ContractResult<HandleResponse> = ContractResult::Ok(HandleResponse::default());
   |                                                                         ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::default::Default` is not implemented for `types::Empty`
   |
   = note: required because of the requirements on the impl of `std::default::Default` for `results::handle::HandleResponse`
   = note: required by `std::default::Default::default`
```

Reply:
Thanks for checking